tests/test_location_pi.py

```diff
@@ -0,0 +1,508 @@
+                self.test_location_01._process_pi_count_moves(count_moves)
+
+    #
+    ## PI Inventory Adjustments
+    #
+
+    def test11__process_inventory_adjustments_returns_None_when_no_drift(self):
+        """ Returns None when no request item is specified """
+        inv = self.test_location_01._process_inventory_adjustments([])
+
+        self.assertIsNone(inv, "Returned a new inventory for empty request")
+
+    def test12_get_stock_drift_no_package_token(self):
+        """
+        Returns the correct metadata for the picking creation if
+        a request item contains a 'NO_PACKAGE' package.
+        """
+        self.create_quant(self.apple.id, self.test_location_01.id, 4)
+        adjs_req = [{"product_id": self.apple.id,
+                     "package_name": "tes12_NO_PACKAGE",
+                     "quantity": 8}]
+
+        stock_drift = self.test_location_01._get_stock_drift(adjs_req)
+
+        self.assertEqual(len(stock_drift), 1, "Not a unique stock drift entry")
+
+        (stock_info, qty), = stock_drift.items()
+
+        self.assertEqual(qty, 8, "Wrong adjustment quantity")
+        self.assertEqual(stock_info.product_id, self.apple.id,
+                         "Wrong product id")
+        self.assertFalse(stock_info.package_id, "Wrongly added a product id")
+        self.assertFalse(stock_info.lot_id, "Wrongly added a lot id")
+
+    def test13_get_stock_drift_new_package_token(self):
+        """
+        Returns the correct metadata for the picking creation if
+        a request item contains a 'NEWPACKAGE' package.
+        """
+        Package = self.env['stock.quant.package']
+
+        test_start_date = datetime.now() - self.datetime_tolerance
+        new_package_name = "tes13_NEWPACKAGE"
+        self.create_quant(self.apple.id, self.test_location_01.id, 4)
+        adjs_req = [{"product_id": self.apple.id,
+                     "package_name": new_package_name,
+                     "quantity": 6}]
+
+        stock_drift = self.test_location_01._get_stock_drift(adjs_req)
+
+        self.assertEqual(len(stock_drift), 1, "Not a unique stock drift entry")
+
+        (stock_info, qty), = stock_drift.items()
+
+        self.assertEqual(qty, 6, "Wrong adjustment quantity")
+        self.assertEqual(stock_info.product_id, self.apple.id,
+                         "Wrong product id")
+        self.assertIsNotNone(stock_info.package_id,
+                             "Does not contain a package")
+
+        new_package = Package.get_package(stock_info.package_id)
+        new_package_date = datetime.strptime(new_package.create_date,
+                                             ODOO_DATETIME_FORMAT)
+
+        self.assertTrue(new_package_date >= test_start_date,
+                        "New package does not seem new")
+        self.assertFalse(stock_info.lot_id, "Wrongly added a lot id")
+
+    def test14_get_stock_drift_new_package_token_multiple_products(self):
```

Comment:
I would like the same test but different new packages. Also one test with one package multiple products but we only ajdust one of them.